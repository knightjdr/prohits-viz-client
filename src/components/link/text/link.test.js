import React from 'react';
import { render } from '@testing-library/react';

import Link from './link';

describe('Link style', () => {
  describe('with style for :visited', () => {
    let container;

    beforeAll(() => {
      ({ container } = render(<Link href="/" />));
    });

    it('should match snapshot', () => {
      expect(container).toMatchSnapshot();
    });

    it('should set :visited color', () => {
      const modifier = {
        modifier: '&:visited',
      };
      expect(container.firstChild).toHaveStyleRule('color', '#1E6C60', modifier);
    });
  });

  describe('show outline on :focus', () => {
    let container;

    beforeAll(() => {
      ({ container } = render(<Link href="/" />));
    });

    it('should add outline on focus', () => {
      const modifier = {
        modifier: '&:focus',
      };
      expect(container.firstChild).toHaveStyleRule('outline', 'auto', modifier);
    });

    it('should set outline properties on focus', () => {
      const modifier = {
        modifier: '&:focus',
      };
      expect(container.firstChild).toHaveStyleRule('outline-color', '#4D90FE', modifier);
    });
  });

  describe('with no style for :visited', () => {
    let container;

    beforeAll(() => {
      ({ container } = render(
        <Link
          href="/"
          visited={false}
        />,
      ));
    });

    it('should match snapshot', () => {
      expect(container).toMatchSnapshot();
    });

    it('should unset :visited color', () => {
      const modifier = {
        modifier: '&:visited',
      };
      expect(container.firstChild).toHaveStyleRule('color', '#2468e5', modifier);
    });
  });

  describe('no outline on :focus', () => {
    let container;

    beforeAll(() => {
      ({ container } = render(
        <Link
          href="/"
          outline={false}
        />,
      ));
    });

    it('should add outline on focus', () => {
      const modifier = {
        modifier: '&:focus',
      };
      expect(container.firstChild).toHaveStyleRule('outline', 'none', modifier);
    });

    it('should set outline properties on focus', () => {
      const modifier = {
        modifier: '&:focus',
      };
      expect(container.firstChild).toHaveStyleRule('outline-color', undefined, modifier);
    });
  });
});
