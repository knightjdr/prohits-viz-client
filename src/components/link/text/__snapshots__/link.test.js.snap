// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link style with no style for :visited should match snapshot 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
}

.c0:link {
  color: #2468e5;
}

.c0:visited {
  color: #2468e5;
}

.c0:hover,
.c0:active {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0:focus {
  outline: auto;
  outline-color: #4D90FE;
  outline-offset: 2px;
  outline-style: dashed;
  outline-width: 2px;
}

<div>
  <a
    class="c0"
    href="/"
  />
</div>
`;

exports[`Link style with style for :visited should match snapshot 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
}

.c0:link {
  color: #2468e5;
}

.c0:visited {
  color: #1E6C60;
}

.c0:hover,
.c0:active {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0:focus {
  outline: auto;
  outline-color: #4D90FE;
  outline-offset: 2px;
  outline-style: dashed;
  outline-width: 2px;
}

<div>
  <a
    class="c0"
    href="/"
  />
</div>
`;
